Documentation/spec/docfx_document_schema.md

```diff
@@ -0,0 +1,197 @@
+
+DocFX Document Schema (abbreviated to `this schema` below) is introduced to address ths problem. This schema is a JSON media type for defining the structure of a DocFX document. This schema is intended to define manipulation, documentation, validation of the document data.
+
+## 2. Conventions and Terminology
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## 3. Overview
+DocFX Document Schema is in [JSON](http://www.json.org/) format. It borrows most syntax from [JSON Schema](http://json-schema.org/), while it also introduces some other syntax to manuplating the data.
+
+### 3.1 Validation
+This schema can describe the structure of a DocFX document. 
+[JSON schema validation](http://json-schema.org/latest/json-schema-validation.html) already defines many keywords. This schema starts from supporting limited keyword like `type`, `properties`.
+
+### 3.2 Manipulation
+This schema can describe how to handle each property of the document.
+
+Some handler is predefined in DocFX, like markup a string using [DFM](..\spec\docfx_flavored_markdown.md), mark a property as an uid, include content from another source, etc. 
+
+Some handler can be injected into the schema parsing process to accomplish some custom steps.
+
+### 3.3 Integration with DocFX
+Schema is passed to DocFX as part of the template. It MUST be put in the `schemas` sub folder under template folder. The file name is treated as the schema name, also known as the document type of a template.
+
+For example, to add a schema for a document type `ManagedReference`, a schema file should added like `{templateFolder}\schemas\ManagedReference.json`.
+
+Schema describes the object model. The format of the object model is not constrained. YAML and JSON should both be supported.
+
+## 4. General Considerations
+* This schema will reuse keywords in JSON schema, but will not change the defition of the existing one.
+* This schema can validate and manipulate single elements in place, but will not reconstruct the object.
+
+## 5. Validation keywords
+
+### 5.1 type
+
+Same as in JSON schema: http://json-schema.org/latest/json-schema-validation.html#rfc.section.6.25
+
+> The value of this keyword MUST be either a string or an array. If it is an array, elements of the array MUST be strings and MUST be unique.
+>
+> String values MUST be one of the six primitive types ("null", "boolean", "object", "array", "number", or "string"), or "integer" which matches any number with a zero fractional part.
+>
+> An instance validates if and only if the instance is in any of the sets listed for this keyword.
```

Comment:
add a todo...

---
In reply to: [122871633](https://github.com/dotnet/docfx/pull/1771#discussion_r122871633) [](ancestors = 122871633)